File: file-keeper/src/api/processes.ts
import { invoke } from '@tauri-apps/api/core'
import type { ProcessInfo } from '../types/process'

export async function findFileProcesses(filePath: string): Promise<ProcessInfo[]> {
  return await invoke<ProcessInfo[]>('find_file_processes', { filePath })
}

export async function closeProcess(windowHandle: number): Promise<void> {
  return await invoke<void>('close_app_process', { windowHandle })
}

export async function closeFileProcesses(filePath: string): Promise<number> {
  return await invoke<number>('close_file_processes', { filePath })
}

File: file-keeper/src-tauri/src/commands/processes.rs
use crate::platform::ProcessInfo;
use crate::utils::process_matcher;

#[tauri::command]
pub async fn find_file_processes(
    file_path: String,
) -> Result<Vec<ProcessInfo>, String> {
    process_matcher::find_processes_for_file(&file_path)
}

#[tauri::command]
pub async fn close_process(
    pid: u32,
) -> Result<(), String> {
    process_matcher::kill_process(pid)
}

#[tauri::command]
pub async fn close_file_processes(
    file_path: String,
) -> Result<usize, String> {
    let processes = process_matcher::find_processes_for_file(&file_path)?;
    let count = processes.len();

    for process in processes {
        process_matcher::kill_process(process.pid)?;
    }
    Ok(count)
}
